internal/common/spiretrustsource.go

```diff
@@ -113,6 +111,16 @@ func NewSpireTrustSource(spireEndpointURLs map[string]string, localBackupDir str
 	return source, nil
 }
 
+// NewSpireTestSource creates a new spire trust source with the test flag on.
+func NewSpireTestSource(spireEndpointURLs map[string]string, localBackupDir string) (*SpireTrustSource, error) {
+	ts, err := NewSpireTrustSource(spireEndpointURLs, localBackupDir)
+	if err != nil {
+		return nil, err
+	}
+	ts.testing = true
+	return ts, nil
+}
+
```

Comment:
I don't love this. I'm going to give some thought to a better way to accomplish what's needed.

Reply:
OK. I would love to use method overriding to add an optional testing flag, but Go doesn't have overriding.

Reply:
Another common pattern is to pass in a config struct that can take on default values instead of using individual parameters.

Reply:
You could also pass an optional configuration function ala https://godoc.org/github.com/pkg/term#CBreakMode

Reply:
@dennisgove seemed to have the main problem of "we don't want to have test flags in prod code", so I don't think extra configuration functions would fix that problem. I'm really not sure how to fix that problem other than making a whole new type of watcher for testing purposes, but I feel like that defeats the purpose of the tests.